src/parser/tests/class_parser_tests.rs

```diff
@@ -0,0 +1,225 @@
+use crate::{
+    ast::*,
+    parser::{parse, tests::lex},
+};
+
+#[test]
+fn simple_class_with_defaults_can_be_parsed() {
+    let lexer = lex("CLASS MyClass END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+
+    assert_eq!(class.name, "MyClass");
+    assert_eq!(class.poly_mode, Some(PolymorphismMode::None));
+    assert_eq!(unit.implementations.len(), 0);
+}
+
+#[test]
+fn simple_class_can_be_parsed() {
+    let lexer = lex("CLASS ABSTRACT MyClass END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+
+    assert_eq!(class.name, "MyClass");
+    assert_eq!(class.poly_mode, Some(PolymorphismMode::Abstract));
+    assert_eq!(unit.implementations.len(), 0);
+}
+
+#[test]
+fn simple_class2_can_be_parsed() {
+    let lexer = lex("CLASS FINAL MyClass2 END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+
+    assert_eq!(class.name, "MyClass2");
+    assert_eq!(class.poly_mode, Some(PolymorphismMode::Final));
+    assert_eq!(unit.implementations.len(), 0);
+}
+
+#[test]
+fn method_with_defaults_can_be_parsed() {
+    let lexer = lex("CLASS MyClass METHOD testMethod END_METHOD END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+    assert_eq!(unit.implementations.len(), 1);
+
+    let method_pou = &unit.units[1];
+    assert_eq!(method_pou.pou_type, PouType::Method);
+    let method = &unit.implementations[0];
+
+    assert_eq!(method_pou.name, "testMethod");
+    assert_eq!(method.access, Some(AccessModifier::Protected));
+    assert_eq!(method_pou.poly_mode, Some(PolymorphismMode::None));
+    assert_eq!(method_pou.return_type, None);
+    assert_eq!(method.overriding, false);
+}
+
+#[test]
+fn method_can_be_parsed() {
+    let lexer =
+        lex("CLASS MyClass METHOD INTERNAL FINAL OVERRIDE testMethod2 END_METHOD END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+    assert_eq!(unit.implementations.len(), 1);
+
+    let method_pou = &unit.units[1];
+    assert_eq!(method_pou.pou_type, PouType::Method);
+    let method = &unit.implementations[0];
+
+    assert_eq!(method_pou.name, "testMethod2");
+    assert_eq!(method.access, Some(AccessModifier::Internal));
+    assert_eq!(method_pou.poly_mode, Some(PolymorphismMode::Final));
+    assert_eq!(method_pou.return_type, None);
+    assert_eq!(method.overriding, true);
+}
+
+#[test]
+fn two_methods_can_be_parsed() {
```

Comment:
We also need a test for methods on function blocks.

Reply:
Are you happy with those I added now?

Reply:
Looks good